mango-python/bdgenomics/mango/variantViz.py

```diff
@@ -0,0 +1,178 @@
+#
+# Licensed to Big Data Genomics (BDG) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The BDG licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from collections import Counter, OrderedDict
+from cigar import Cigar
+import matplotlib.pyplot as plt
+import numpy as np
+import matplotlib.ticker as ticker
+import math
+from pyspark.sql import Row
+plt.rcdefaults()
+plt.rcParams['figure.figsize'] = [10, 8]
+
+default_bins=100
+
+class VariantsGenomicRegion(object):
+    """
+    Plot variant density across chromosome region
+    """
+    def __init__(self,ss,variantRDD, start, end, contigName, binSize=1000000):
+        """
+        :param ss: SparkContext 
+        :param variantRDD: bdgenomics variantRDD
+        :param start: genomic start position
+        :param end:  genomic stop position
+        :param contigName: contig name
+        :param binSize: region bin size defaults to 1000000  
+        """
+        data = variantRDD.toDF().filter(
+            (variantRDD.toDF().start > start) &
+            (variantRDD.toDF().start < end) &
+            (variantRDD.toDF().contigName == contigName) ).rdd.map(
+              lambda w: w.asDict()).map(
+               lambda x: (x['contigName'], int(math.floor(float(x['start'])/binSize)))).countByValue()
+        self.data = data
+        self.contigName=contigName
+    def plot(self):
+        x = [ z[0][1] for z in self.data.items() ]
```

Comment:
We now return plot objects so the user can mofify them, instead of plotting. see https://github.com/bigdatagenomics/mango/blob/master/mango-python/bdgenomics/mango/alignments.py#L307